source/val/construct.cpp

```diff
@@ -64,4 +67,52 @@ const BasicBlock* Construct::exit_block() const { return exit_block_; }
 BasicBlock* Construct::exit_block() { return exit_block_; }
 
 void Construct::set_exit(BasicBlock* block) { exit_block_ = block; }
+
+Construct::ConstructBlockSet Construct::blocks(Function* function) const {
+  auto header = entry_block();
+  auto merge = exit_block();
```

Comment:
Since a precondition for this function is that the exit block is set, can we assert that `merge != nullptr`?  Or are asserts not done in the validator?

Reply:
Added an assert for header and merge.